### PR TITLE
    [REF-6034] AUCScore: Adding MLOps Classification Metrics - AUC Score

### PR DESCRIPTION
**Pull request by harshilpm**
[REF-6034] AUCScore: Adding MLOps Classification Metrics - AUC Score

    Usages:
        labels_pred = [1, 0, 1, 1, 1, 0]
        labels_actual = [0, 1, 0, 0, 0, 1]

        fpr, tpr, thresholds = sklearn.metrics.roc_curve(labels_actual, labels_pred, pos_label=2)
        auc = sklearn.metrics.auc(fpr, tpr)

        # first way
        pm.set_stat(ClassificationMetrics.AUC, auc)

        # second way
        pm.metrics.auc(x=fpr, y=tpr)

    Testing Done
    - Unit Test

### CODE REVIEW

**Review by lior-amar-pm — APPROVED** [review]

File: mlops/parallelm/mlops/stats/stats_helper.py [hunk 1]
```diff
@@ -55,6 +56,11 @@ def _set_classification_stat(self, name, data, model_id, timestamp, **kwargs):
                 AccuracyScore.get_mlops_accuracy_stat_object(accuracy_score=data)
             category = StatCategory.TIME_SERIES
 
+        elif name == ClassificationMetrics.AUC:
```
Comment (lior-amar-pm) [inline mlops/parallelm/mlops/stats/stats_helper.py:59]: The big if/elif/elif... will be replaces with registering a handler function of each stats. This will come in a separate PR after the initial 12 stats goes in.

File: mlops/parallelm/mlops/ml_metrics_stat/auc.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+from parallelm.mlops.metrics_constants import ClassificationMetrics
+from parallelm.mlops.mlops_exception import MLOpsStatisticsException
+from parallelm.mlops.stats.single_value import SingleValue
+from parallelm.mlops.stats_category import StatCategory
+
+
+class AUC(object):
```
Comment (lior-amar-pm) [inline mlops/parallelm/mlops/ml_metrics_stat/auc.py:7]: This class will be removed (since it is repeating itself for most of the stats.. 
The cleanup will come in the 13 PR after the inital 12 metric goes in.

**Review by lior-amar-pm — APPROVED** [review]

**Review by yakov-g — APPROVED** [review]

File: mlops/parallelm/mlops/stats/stats_helper.py (hunk 1 above)
Comment (harshilpm) [reply mlops/parallelm/mlops/stats/stats_helper.py:59]: yes. i had put them in registry :)
pr: https://github.com/mlpiper/mlpiper/pull/93

File: mlops/parallelm/mlops/ml_metrics_stat/auc.py (hunk 2 above)
Comment (harshilpm) [reply mlops/parallelm/mlops/ml_metrics_stat/auc.py:7]: cleanup/beautiy pr: https://github.com/mlpiper/mlpiper/pull/93

**Comment by harshilpm** [comment]
closing pr .. please review https://github.com/mlpiper/mlpiper/pull/95
